File: Fresh/Fresh.Model/Requests/ProductInsertRequest.cs
using System.ComponentModel.DataAnnotations;

namespace Fresh.Model.Requests
{
    public class ProductInsertRequest
    {
        [Required(ErrorMessage = "This filed is required !")]
        public string Name { get; set; }

        public byte[]? Image { get; set; }

        [Required(ErrorMessage = "This filed is required !")]
        public UnitType Unit { get; set; }

        [Required(ErrorMessage = "This filed is required !")]
        public bool IsActive { get; set; } = true;

        [Required(ErrorMessage = "This filed is required !")]
        public int ProductTypeId { get; set; }
    }
}

File: Fresh/UI/fresh_app/src/app/model/requests/ProductInsertRequest.ts
import { UnitType } from "../UnitType";

export class ProductInsertRequest {
    name: string;
    image?: string;
    unit: UnitType;
    isActive: boolean;
    productTypeId: number;

    constructor(name: string, image: string | undefined, unit: UnitType, isActive: boolean, productTypeId: number) {
        this.name = name;
        this.image = image;
        this.unit = unit;
        this.isActive = isActive;
        this.productTypeId = productTypeId;
    }
}